src/fetch.ts

```diff
@@ -31,8 +32,21 @@ export function createFetch ({ fetch }: CreateFetchOptions): $Fetch {
       if (opts.params) {
         request = withQuery(request, opts.params)
       }
+      if (opts.body && typeof opts.body === 'object' && ['patch', 'post', 'put'].includes(opts.method?.toLowerCase() || '')) {
+        try {
+          opts.body = JSON.stringify(opts.body)
+          opts.headers = opts.headers || {}
+          if (Array.isArray(opts.headers)) {
+            opts.headers.push(['content-type', 'application/json'])
+          } else if ('set' in opts.headers) {
+            ;(opts.headers as Headers).set('content-type', 'application/json')
+          } else {
+            opts.headers['content-type'] = 'application/json'
+          }
+        } catch {}
```

Comment:
try/catch scope is wider than necessary (i think purpose was to handle JSON.stringify issues) but i think for that we may either keep original error or fallback to opts.body